### PR TITLE
wip: Reduce ARC traffic with inout wrapper

### PR DESCRIPTION
**Pull request by simonjbeaumont**
_[One line description of your change]_

### Motivation:

_[Explain here the context, and why you're making that change. What is the problem you're trying to solve.]_

### Modifications:

_[Describe the modifications you've done.]_

### Result:

_[After your change, what will change.]_

### CODE REVIEW

**Comment by simonjbeaumont** [comment]
@swift-nio-bot test perf please

**Comment by simonjbeaumont** [comment]
@swift-nio-bot test perf please

File: Sources/NIOCore/ChannelPipeline.swift [hunk 1]
```diff
@@ -1503,7 +1512,7 @@ public final class ChannelHandlerContext: ChannelInvoker {
 
     /// Signal to the next `ChannelHandler` that a read burst has finished.
     public func fireChannelReadComplete() {
-        self.next?.invokeChannelReadComplete()
+        executeIfNotNil(&self.next) { $0.invokeChannelReadComplete() }
```
Comment (Lukasa) [inline Sources/NIOCore/ChannelPipeline.swift:1515]: Sadly I don't think we can make this optimisation: it'll violate the law of exclusivity. It's possible for us to synchronously reconfigure the pipeline, which will write to this property.

Comment (simonjbeaumont) [reply Sources/NIOCore/ChannelPipeline.swift:1515]: Yeah, I was looking at something referenced in an old issue which suggested this is something we could try although I can see that when I apply this particular variant across some tests they explode with access issues.

I'm studying the other issue though and think I just overlooked something and there may be something we can do here.

**Pull request closed without merging**